app/controller/redactor_rails/documents_controller.rb

```diff
@@ -8,23 +8,61 @@ def index
   end
 
   def create
-    @document = RedactorRails.document_model.new
+    if params[:version_4].present?
+      # Redactor 4
+      process_version_4_document
+    else
+      # Legacy Redactor
+      process_version_2_document
+    end
 
-    file = params[:file]
-    @document.data = RedactorRails::Http.normalize_param(file, request)
-    if @document.has_attribute?(:"#{RedactorRails.devise_user_key}")
-      @document.send("#{RedactorRails.devise_user}=", redactor_current_user)
-      @document.assetable = redactor_current_user
+  end
+
+  private
+
+  def process_version_4_document
+    files = params[:file]
+    results = {}
+    file_counter = 0
+
+    files.each do |file|
+      @document = RedactorRails.document_model.new
+      attach_document(file)
+
+      if @document.save
+        file_counter += 1
```

Comment:
Do we need to increment `file_counter` outside of the `@document.save` block?

I'm just thinking through _"**Image #1** worked, **Image #2** failed, and then **Image #3** succeeded after that..."_

I'm wondering if only incremeneting the `file_counter` index on _success_ might mess up that flow (if it's even possible in the first place)

Reply:
Yeah, I could see that. It wouldn't hurt to increment it regardless.